src/panels/lovelace/cards/hui-shopping-list-card.ts

```diff
@@ -72,38 +68,73 @@ class HuiShoppingListCard extends hassLocalizeLitMixin(LitElement)
   }
 
   protected render(): TemplateResult {
-    if (!this._config || !this._hass) {
+    if (!this._config || !this.hass) {
       return html``;
     }
 
     return html`
       ${this.renderStyle()}
       <ha-card .header="${this._config.title}">
-      ${repeat(
-        this._items!,
-        (item) => item.id,
-        (item, index) =>
-          html`
-        <div class="editRow">
-          <paper-checkbox
-            slot="item-icon"
-            id=${index}
-            ?checked=${item.complete}
-            .itemId=${item.id}
-            @click=${this._completeItem}
-            tabindex='0'
-          ></paper-checkbox>
-          <paper-item-body>
-            <paper-input
-              no-label-float
-              .value='${item.name}'
-              .itemId=${item.id}
-              @change=${this._saveEdit}
-            ></paper-input>
-          </paper-item-body>
+        ${
+          repeat(
+            this._items!.filter((item) => !item.complete),
+            (item) => item.id,
+            (item, index) =>
+              html`
+                <div class="editRow">
+                  <paper-checkbox
+                    slot="item-icon"
+                    id="${index}"
+                    ?checked="${item.complete}"
+                    .itemId="${item.id}"
+                    @click="${this._completeItem}"
+                    tabindex="0"
+                  ></paper-checkbox>
+                  <paper-item-body>
+                    <paper-input
+                      no-label-float
+                      value="${item.name}"
+                      .itemId="${item.id}"
+                      @change="${this._saveEdit}"
+                    ></paper-input>
+                  </paper-item-body>
+                </div>
+              `
+          )
+        }
+        <div class="divider"></div>
```

Comment:
We should not show this divider and label if there are no checked items. 

I suggest that instead of filtering the items twice, you go over it once and create 2 lists.

Reply:
![image](https://user-images.githubusercontent.com/1444314/48187079-56e11500-e33a-11e8-8c31-c2ce7139b874.png)